aptos-move/framework/aptos-framework/sources/transaction_context.move

```diff
@@ -46,6 +57,128 @@ module aptos_framework::transaction_context {
+    /// This function aborts if called outside of the transaction prologue, execution, or epilogue phases.
+    native fun gas_unit_price_internal(): u64;
+    public fun gas_unit_price(): u64 {
+        assert!(features::transaction_context_extension_enabled(), error::invalid_state(ETRANSACTION_CONTEXT_EXTENSION_NOT_ENABLED));
+        gas_unit_price_internal()
+    }
+
+    /// Return the chain ID specified for the current transaction.
+    /// This function aborts if called outside of the transaction prologue, execution, or epilogue phases.
+    native fun chain_id_internal(): u8;
+    public fun chain_id(): u8 {
+        assert!(features::transaction_context_extension_enabled(), error::invalid_state(ETRANSACTION_CONTEXT_EXTENSION_NOT_ENABLED));
+        chain_id_internal()
+    }
+
+    struct EntryFunctionPayload has copy, drop {
+        account_address: address,
+        module_name: String,
+        function_name: String,
+        ty_args_names: vector<String>,
+        args: vector<vector<u8>>,
+    }
+
+    /// Return the entry function payload if the current transaction has such a payload. Otherwise, return `None`.
+    /// This function aborts if called outside of the transaction prologue, execution, or epilogue phases.
+    native fun entry_function_payload_internal(): Option<EntryFunctionPayload>;
+    public fun entry_function_payload(): Option<EntryFunctionPayload> {
+        assert!(features::transaction_context_extension_enabled(), error::invalid_state(ETRANSACTION_CONTEXT_EXTENSION_NOT_ENABLED));
+        entry_function_payload_internal()
+    }
+
+    /// Return the account address of the entry function payload.
+    public fun account_address(payload: &EntryFunctionPayload): address {
+        assert!(features::transaction_context_extension_enabled(), error::invalid_state(ETRANSACTION_CONTEXT_EXTENSION_NOT_ENABLED));
+        payload.account_address
+    }
+
+    /// Return the module name of the entry function payload.
+    public fun module_name(payload: &EntryFunctionPayload): String {
+        assert!(features::transaction_context_extension_enabled(), error::invalid_state(ETRANSACTION_CONTEXT_EXTENSION_NOT_ENABLED));
+        payload.module_name
+    }
+
+    /// Return the function name of the entry function payload.
+    public fun function_name(payload: &EntryFunctionPayload): String {
+        assert!(features::transaction_context_extension_enabled(), error::invalid_state(ETRANSACTION_CONTEXT_EXTENSION_NOT_ENABLED));
+        payload.function_name
+    }
+
+    /// Return the type arguments names of the entry function payload.
+    public fun type_arg_names(payload: &EntryFunctionPayload): vector<String> {
+        assert!(features::transaction_context_extension_enabled(), error::invalid_state(ETRANSACTION_CONTEXT_EXTENSION_NOT_ENABLED));
+        payload.ty_args_names
+    }
+
+    /// Return the arguments of the entry function payload.
+    public fun args(payload: &EntryFunctionPayload): vector<vector<u8>> {
```

Comment:
Is this needed only for the prologue/epilogue? I would assume that the entry function itself already gets these as values so it's no use in that context.

Reply:
Right, they may not be useful at the entry point of the entry function execution. They may be useful downstream in the function call chain, though.

They are most useful for the prologue and epilogue.